client/src/main/scala/org/apache/celeborn/client/ChangePartitionManager.scala

```diff
@@ -81,6 +84,42 @@ class ChangePartitionManager(
                   new Runnable {
                     override def run(): Unit = {
                       requests.synchronized {
+                        // Move all pending requests to requests
+                        pendingChangePartitionRequests.get(shuffleId).synchronized {
```

Comment:
Since `changePartitionRequests` will not add change partition requests, do we really want `pendingChangePartitionRequests`? Maybe we can just add the checking latest partition logic in original code.